pennylane/devices/default_qubit.py

```diff
@@ -185,6 +186,25 @@ def __init__(
             "Toffoli": self._apply_toffoli,
         }
 
+    @property
+    def stopping_condition(self):
+        def accepts_obj(obj):
+            if getattr(obj, "has_matrix", False):
+                # pow operations dont work with backprop or adjoint without decomposition
+                # use class name string so we don't need to use isisntance check
+                if obj.__class__.__name__ == "Pow" and qml.operation.is_trainable(obj):
+                    return False
+                return True
+            if obj.name in self.observables:  # not all observables have matrices
+                return True
+            with warnings.catch_warnings():  # state prep routines
+                warnings.simplefilter("ignore")
+                if obj.name in self.operations:
+                    return True
```

Comment:
I was trying to turn `operations` into a class property that raised a warning when accessed.  Unfortunately, too many places (tests primarily) try to dynamically manipulate the `dev.operations` set, which would be impossible with a class property. I didn't want to have to rewrite a bunch of tests, especially with the time crunch.

I should probably get rid of this warnings logic again.

We have a couple of supported operations (state prep and snapshot) that are supported but don't have matrices.

Reply:
> I should probably get rid of this warnings logic again.

Yep, let's remove the warning, since:

- we are rethinking the class anyway
- this could cause the warning to arise for users just due to `operations` being accessed internally somewhere.